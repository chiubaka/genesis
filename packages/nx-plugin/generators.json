{
  "$schema": "http://json-schema.org/schema",
  "name": "nx-plugin",
  "version": "0.0.1",
  "generators": {
    "eslint": {
      "factory": "./src/generators/linting/eslint",
      "schema": "./src/generators/linting/eslint/schema.json",
      "description": "Generates ESLint configuration and setup"
    },
    "gitHooks": {
      "factory": "./src/generators/gitHooks",
      "schema": "./src/generators/gitHooks/schema.json",
      "description": "Generates a git hooks setup"
    },
    "linting": {
      "factory": "./src/generators/linting",
      "schema": "./src/generators/linting/schema.json",
      "description": "Generates linting setup including EsLint and Prettier"
    },
    "preset": {
      "factory": "./src/generators/preset",
      "schema": "./src/generators/preset/schema.json",
      "description": "Generates a full workspace with batteries included!"
    },
    "prettier": {
      "factory": "./src/generators/linting/prettier",
      "schema": "./src/generators/linting/prettier/schema.json",
      "description": "Generates Prettier configuration and setup"
    }
  }
}
