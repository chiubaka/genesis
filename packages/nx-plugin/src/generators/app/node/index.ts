export * from "./nodeAppGenerator";
export { nodeAppGenerator as default } from "./nodeAppGenerator";
