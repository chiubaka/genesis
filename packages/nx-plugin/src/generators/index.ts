export * from "./linting";
export * from "./preset";
