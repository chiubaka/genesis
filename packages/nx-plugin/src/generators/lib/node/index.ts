export * from "./nodeLibGenerator";
export { nodeLibGenerator as default } from "./nodeLibGenerator";
