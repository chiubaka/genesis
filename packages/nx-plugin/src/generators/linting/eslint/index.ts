export * from "./eslintGenerator";
export { eslintGenerator as default } from "./eslintGenerator";
