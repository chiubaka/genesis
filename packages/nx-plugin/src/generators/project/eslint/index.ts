export * from "./eslintProjectGenerator";
export { eslintProjectGenerator as default } from "./eslintProjectGenerator";
