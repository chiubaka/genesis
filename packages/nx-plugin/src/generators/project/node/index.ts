export * from "./nodeProjectGenerator";
export { nodeProjectGenerator as default } from "./nodeProjectGenerator";
