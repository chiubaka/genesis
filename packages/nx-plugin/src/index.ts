export * from "./generators";
export * from "./utils";
