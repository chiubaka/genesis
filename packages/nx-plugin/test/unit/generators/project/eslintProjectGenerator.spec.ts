import { readJson, Tree } from "@nrwl/devkit";
import { createTreeWithEmptyWorkspace } from "@nrwl/devkit/testing";
import { libraryGenerator } from "@nrwl/node";
import { Linter } from "eslint";

import { eslintProjectGenerator } from "../../../../src/generators";
import { Project } from "../../../utils";

describe("eslintProjectGenerator", () => {
  let tree: Tree;
  let project: Project;

  beforeAll(async () => {
    const projectName = "eslint-project";
    const projectType = "library";

    tree = createTreeWithEmptyWorkspace();
    await libraryGenerator(tree, {
      name: projectName,
      compiler: "tsc",
    });
    eslintProjectGenerator(tree, {
      projectName,
      projectType,
    });

    project = new Project(tree, projectName, projectType);
  });

  it("generates a .eslintrc.json file", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.path(".eslintrc.json"))).toBe(true);
  });

  describe(".eslintrc.json", () => {
    let eslintConfig: Linter.Config;

    beforeAll(() => {
      eslintConfig = readJson<Linter.Config>(
        tree,
        project.path(".eslintrc.json"),
      );
    });

    it("extends from the root monorepo .eslintrc.json file", () => {
      expect(eslintConfig.extends).toEqual(["../../.eslintrc.json"]);
    });

    it("only ignores node_modules", () => {
      expect(eslintConfig.ignorePatterns).toEqual(["node_modules"]);
    });
  });
});
