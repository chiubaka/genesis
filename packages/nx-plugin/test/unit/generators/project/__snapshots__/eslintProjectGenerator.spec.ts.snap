// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslintProjectGenerator .eslintrc.json matches snapshot 1`] = `
{
  "extends": [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": [
    "node_modules",
  ],
  "overrides": [
    {
      "files": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "parserOptions": {
        "project": [
          "packages/eslint-project/tsconfig.spec.json",
        ],
      },
    },
    {
      "excludedFiles": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "files": [
        "*.ts",
      ],
      "parserOptions": {
        "project": [
          "packages/eslint-project/tsconfig.lib.json",
        ],
      },
    },
  ],
}
`;
